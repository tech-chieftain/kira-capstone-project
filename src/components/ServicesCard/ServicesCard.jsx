import React, { useState } from "react";
import PropTypes from "prop-types";
import { Card, Col, Row } from "react-bootstrap";
import { useTranslation } from "next-i18next";
import { FaHeart, FaUserCircle } from "react-icons/fa";
import { StarIcon, Button, CardBody, Img } from "./ServicesCard.styled";

const ServicesCard = ({
  person,
  width,
  photo,
  profilePic,
  title,
  description,
  rating,
  price,
  name,
}) => {
  const [favorate, setFavorate] = useState(false);

  const favorated = () => {
    setFavorate(!favorate);
  };

  const { t } = useTranslation("card");

  return (
    <Card style={{ width: `${width}rem` }} className="shadow z-depth-5">
      <Card.Img variant="top" src={photo} alt="" />
      <CardBody>
        {person && (
          <Row md={2} className="d-flex align-items-center mb-2">
            <Col>
              {profilePic ? (
                <Img src={profilePic} roundedCircle fluid />
              ) : (
                <FaUserCircle size="28px" className="text-secondary" />
              )}
            </Col>
            <Col className="username">
              <Card.Subtitle>{name}</Card.Subtitle>
            </Col>
          </Row>
        )}
        <Card.Text>{title}</Card.Text>
        <Card.Text>{description}</Card.Text>
      </CardBody>
      <Card.Footer className="bg-white d-flex align-items-center justify-content-between">
        <small className="text-muted">
          {t("card.startingAt")} <span className="h6">${price}</span>
        </small>
      </Card.Footer>
    </Card>
  );
};

ServicesCard.propTypes = {
  person: PropTypes.bool,
  width: PropTypes.number,
  price: PropTypes.number,
  rating: PropTypes.number,
  photo: PropTypes.string,
  description: PropTypes.string,
  title: PropTypes.string,
  name: PropTypes.string,
  profilePic: PropTypes.string,
};

export default ServicesCard;
