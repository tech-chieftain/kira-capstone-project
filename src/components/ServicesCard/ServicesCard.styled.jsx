import styled from "styled-components";
import { FaStar } from "react-icons/fa";

export const StarIcon = styled(FaStar)`
  margin-right: 5px;
  width: 16px;
  height: 16px;
`;

export const Button = styled.button`
  background: transparent;
  border: none;
  color: grey;
  padding: 0;
`;
