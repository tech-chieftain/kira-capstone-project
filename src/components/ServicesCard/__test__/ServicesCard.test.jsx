import React from "react";
import renderer from "react-test-renderer";
import ServicesCard from "../ServicesCard";

test("properly renders the ServicesCard Component", () => {
  const tree = renderer.create(<ServicesCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
