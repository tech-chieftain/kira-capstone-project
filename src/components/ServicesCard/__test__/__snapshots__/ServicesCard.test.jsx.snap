// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly renders the ServicesCard Component 1`] = `
<div
  className="shadow z-depth-5 card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    alt=""
    className="card-img-top"
    src="https://picsum.photos/307/186"
  />
  <div
    className="ServicesCardstyled__CardBody-sc-1wnbk5s-0 jsoemc card-body"
  >
    <div
      className="d-flex align-items-center mb-2 row row-cols-md-2"
    >
      <div
        className="col"
      >
        <img
          className="ServicesCardstyled__Img-sc-1wnbk5s-1 eNnsBq img-fluid rounded-circle"
          src="https://picsum.photos/307/186"
        />
      </div>
      <div
        className="username col"
      >
        <div
          className="card-subtitle h6"
        >
          Username
        </div>
      </div>
    </div>
    <p
      className="card-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
    <small
      className="text-warning d-flex align-items-center"
    >
      <svg
        className="ServicesCardstyled__StarIcon-sc-1wnbk5s-2 eYTJaB"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 576 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
        />
      </svg>
       
      10
    </small>
  </div>
  <div
    className="bg-white d-flex align-items-center justify-content-between card-footer"
  >
    <button
      className="ServicesCardstyled__Button-sc-1wnbk5s-3 fa-DOSR"
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
        />
      </svg>
    </button>
    <small
      className="text-muted"
    >
      STARTING AT 
      <span
        className="h6"
      >
        $
        8
      </span>
    </small>
  </div>
</div>
`;
